src/machine/program.cpp

```diff
@@ -49,9 +48,9 @@ void program::reserve_stacks()
 // Constructors.
 //-----------------------------------------------------------------------------
 
-program::program()
+program::program(const settings& settings)
   : script_(default_script_),
-    transaction_(default_tx_),
+    transaction_(chain::transaction(settings)),
```

Comment:
Here, we're binding a temporary to a const reference which will result in a dangling reference after the ctor exits. I propose to change the type of `transaction_` to `std::shared_ptr` to fix this.

Reply:
That would result in a copy of the transaction when the tx is passed. The issue would be resolved by eliminating the settings constructor.

Reply:
Ha, that’s not actually a shared pointer, that’s a weak pointer.

Reply:
I do believe eliminating the members for the chain (and config) classes would be better.

Reply:
OK. I have no idea though how do get rid of the member in config because it's used in `std::istream& operator>>(std::istream& input, output& argument)` so I cannot pass it in as an argument. The only thing that comes to mind is to remove this check
```cpp
    if (amount > argument.settings_.max_money())
    {
        BOOST_THROW_EXCEPTION(invalid_option_value(tuple));
    }
```

Reply:
See previous comment. Eliminate this test.

Reply:
> I do believe eliminating the members for the chain (and config) classes would be better.

Done.